scylla-cdc-base/src/main/java/com/scylladb/cdc/transport/WorkerTransport.java

```diff
@@ -13,5 +13,13 @@
     Map<TaskId, TaskState> getTaskStates(Set<TaskId> tasks);
     void setState(TaskId task, TaskState newState);
     void moveStateToNextWindow(TaskId task, TaskState newState);
-    boolean shouldStop();
+
+    /**
+     * @deprecated Use {@link Worker.stop()} instead")
+     * @return
+     */
+    @Deprecated
+    default boolean shouldStop() {
+        return false;
```

Comment:
Why does a default implementation make any sense here? This method is suppose to change it's value once the stop is requested.

Reply:
No, it is not. Calling stop stops the connector. It is only an implementation detail that this is also a state (other than post - it is off). This is defaulted so clients do not have to implement a deprectated method - they can ignore this and just call Worker::stop when they want to terminate.

Reply:
Yes it is. That was the semantic of this method before your change. If you're changing the semantic then please document what that method means now.

Reply:
?? What is, and what was the semantics? The meaning of the above "shouldStop" has not changed. It is deprecated in favour of the more deterministic Worker::stop (deterministic as in "once you've called and returned from it - is is stopped").